.vscode/typespec.code-snippets

```diff
@@ -0,0 +1,382 @@
+  },
+  "@pollingOperation": {
+    "prefix": "@polling-operation",
+    "body": "@pollingOperation(${1:linkedOperation}, ${2:parameters?})",
+    "description": "Identifies that an operation is a polling operation for an LRO"
+  },
+  "@pollingOperationParameter": {
+    "prefix": "@polling-operation-parameter",
+    "body": "@pollingOperationParameter(${1:targetParameter})",
+    "description": "Used to define how to call custom polling operations for long-running operations"
+  },
+  "@useFinalStateVia": {
+    "prefix": "@use-final-state-via",
+    "body": "@useFinalStateVia(${1:finalState})",
+    "description": "Overrides the final state value for an operation"
+  },
+  "@requestParameter": {
+    "prefix": "@request-parameter",
+    "body": "@requestParameter(${1:propertyName })",
+    "description": "Identifies a property on a request model that serves as a linked operation parameter"
+  },
+  "@responseProperty": {
+    "prefix": "@response-property",
+    "body": "@responseProperty(${1:propertyName})",
+    "description": "Identifies a property on all non-error response models that serve as a linked operation parameter"
+  },
+  "@trait": {
+    "prefix": "@trait",
+    "body": "@trait(${1:traitName})",
+    "description": "`@trait` marks a model type as representing a 'trait' and performs basic validation checks"
+  },
+  "@traitAdded": {
+    "prefix": "@trait-added",
+    "body": "@traitAdded(${1:addedVersion})",
+    "description": "Sets the version for when the trait was added to the specification. Can be applied to either a trait model type or its envelope property"
+  },
+  "@traitContext": {
+    "prefix": "@trait-context",
+    "body": "@traitContext(${1:contexts})",
+    "description": "`@traitContext` sets the applicable context for a trait on its envelope property"
+  },
+  "@traitLocation": {
+    "prefix": "@trait-location",
+    "body": "@traitLocation(${1:contexts})",
+    "description": "`@traitLocation` sets the applicable location for a trait on its envelope property"
+  },
+  "@armCommonTypesVersion": {
+    "prefix": "@arm-common-types-version",
+    "body": "@armCommonTypesVersion(${1:version})",
+    "description": "This decorator is used either on a namespace or a version enum value to indicate the version of the Azure Resource Manager common-types to use for refs in emitted Swagger files"
+  },
+  "@armProviderNamespace": {
+    "prefix": "@arm-provider-namespace",
+    "body": "@armProviderNamespace(${1:providerNamespace})",
+    "description": "`@armProviderNamespace` sets the Azure Resource Manager provider name. It will default to use the Namespace element value unless an override value is specified"
+  },
+  "@armResourceAction": {
+    "prefix": "@arm-resource-action",
+    "body": "@armResourceAction(${1:resourceType})",
+    "description": "`@armResourceAction` marks an operation as an Azure Resource Manager resource action"
+  },
+  "@armResourceCreateOrUpdate": {
```

Comment:
Do we really need to redefine every decorator and template. Doesn't the autocomplete already covers this.


I see a lot of snippet which just gonna drown the user in choices it feels